### DISCUSSION TITLE
Workaround for Chrome tab focus not working in i3 workspaces?

### DISCUSSION THREAD

**Discussion by KevinVillela** [i3/i3#4972] [Q&A] [answered]
Hi i3 gurus,

Background: I use VSCode (Electron-based) inside of Chrome, and it will send notifications whenever the tab is not focused. It detects this with the [window.state.focused](https://code.visualstudio.com/api/references/vscode-api#WindowState) attribute.

Problem: When using i3, if I keep the tab open but switch workspaces, VSCode thinks that it is still focused (i.e. `window.state.focused` is `True`) and won't send me notifications. This is true even if the workspace covers the workspace that VSCode is in. I have tried with and without `compton` as a compositor, but get the same results.

Note that this does work if I switch to a **tabbed** container within the same workspace, although not if I switch to a non-tabbed container in the same workspace. The only workaround I can think of would be to have my "switch workspace" keybinding also change to an empty tabbed container, and when I switch back I could have it open the Chrome tab. Pretty gross, so I'm wondering if anyone else has any ideas, or if perhaps I'm doing something wrong.

Thank you!

**Comment by Airblader** [comment, marked as answer]
From your description it sounds like `window.state.focused` is based on the `_NET_WM_STATE_HIDDEN` atom (which sounds incorrect to me as it has nothing to do with focus). This might actually come from the browser, I think we've seen similar behavior there in the past. Consider filing a bug report for that.

As for a workaround, you can manually set the atom on the window (e.g. using `xprop`), but that's "unstable" as you'll be fighting i3 in setting and removing it. You could also patch i3 to (incorrectly) also set the atom in these scenarios. It's hard to say what side effects this could have, the browser might for example not render updates if the atom is set since it thinks the window isn't visible.

**Comment by KevinVillela** [comment]
Thanks so much for the response! I tried `$ xprop -id 0x4a00005 -f _NET_WM_STATE 32a -set _NET_WM_STATE _NET_WM_STATE_HIDDEN` and it did indeed cause the notification to pop up!

I see what you mean about this workaround being unstable. I've filed https://github.com/microsoft/vscode/issues/149487 to track this, although I'd imagine the actual issue isn't in VSCode but somewhere upstream (like the browser, as you mentioned).
